packages/api-client/src/views/Request/libs/oauth2.ts

```diff
@@ -260,7 +260,9 @@ export const authorizeServers = async (
     }
 
     // Add client id + secret to headers
-    headers.Authorization = `Basic ${btoa(`${flow['x-scalar-client-id']}:${flow.clientSecret}`)}`
+    if (flow.clientSecret) {
```

Comment:
should we check for clientId as well?

Reply:
@amritk Thank you for the review. While I'm not familiar with this codebase, I discovered this fix works for my use case.

Regarding clientId: it appears this will always be present, as line 222 unconditionally sets client_id in formData from `flow['x-scalar-client-id']`. This contrasts with line 230, which checks if flow.clientSecret exists before adding it to formData. 

If my understanding is incorrect, please let me know and I'll happily update the condition.

Reply:
oh sure sorry I didn't look at the rest of the file, guess I missed the context